### PR TITLE
Local metadata validation CLI

### PR DESCRIPTION
**Pull request by knapii-developments**
The PR:

- Provides functionality to validate metadata file against file convention prior to being uploaded.
-  Obtains the latest metadata convention via API call for ‘latest’ metadata convention instead of using ```JSON_CONVENTION``` in scp ingest pipeline.
- Creates a [cli parser](https://github.com/broadinstitute/single_cell_portal/pull/97/files#diff-c329d5699e7b624055ef003727fdbd71) for arguments. Moving passed in arguments made it easier to create tests and is implemented similarly in ingest pipeline. 
- Updates tests for validating metadata convention against a metadata file.
- Updates [requirements.txt](https://github.com/broadinstitute/single_cell_portal/pull/97/files#diff-b4ef698db8ca845e5845c4618278f29a) and config file to use the [latest commit ](https://github.com/broadinstitute/scp-ingest-pipeline/commit/9b8e56e5f3d84668f68c05dcd71208d23e99f96f)for SCP-pipeline as of 5/13/2020

Please make note that a study must exist before metadata validation on the CLI can occur. This is due to [get_study_attribute()](https://github.com/broadinstitute/single_cell_portal/blob/5fa415443a0f93f99e76dd2aedfdf207b8c99cdb/scripts/scp_api.py#L449) not providing dummy values for study accession and dummy BSON objects for the file_id and study_id. Current implementation relies on study name to find the study accession and hard codes dummy values for the file and study ids. This issue is being tracked in [SCP-2421](https://broadworkbench.atlassian.net/browse/SCP-2421)

Command to run (for staging environment) is: ``` python3 manage_study.py upload-metadata --study-name "CLI test" --file ../../scp-ingest-pipeline/tests/data/valid_no_array_v2.0.0.tsv --use-convention ```

The PR satisfies [SCP-2281](https://broadworkbench.atlassian.net/browse/SCP-2281)

### CODE REVIEW

File: scripts/manage_study.py [hunk 1]
```diff
@@ -140,18 +140,24 @@ def download_from_bucket(file_path):
...
+        if verbose:
+            print(f'Study accession {study_accession_res} retrieved for {parsed_args.study_name}')
+        study_accession = study_accession_res.get('study_attribute')
+        metadata = CellMetadata(metadata_path, '', '', study_accession=str(study_accession))
+        convention_res = connection.do_get(command=api_base + 'metadata_schemas/alexandria_convention/latest/json',dry_run=parsed_args.dry_run)
+        if succeeded(convention_res ):
+            if verbose:
+                print(f'Retreieved file for latest metdata convention')
+            convention = convention_res["response"].json()
+            validate_input_metadata(metadata, convention)
```
Comment (jlchang) [inline scripts/manage_study.py:157]: We want to make sure the ingest pipeline code the user has locally is compatible (preferably the same) as the code the validation server (typically production) is using. I'm not 100% sure how we can make sure their local python install of scp-ingest-pipeline is in sync with the version in the docker container that the validation server is using.
 
We could check, at least, that the version of the metadata convention used by ingest pipeline (which should be parsable from IngestPipeline.JSON_CONVENTION) matches the version number of the metadata convention that was retrieved by API (the version number is parsable from the value for $id in the convention json). 

If the conventions versions don't match, the logic/code being run for validation will definitely not match and the user should be prompted to action (for validation against production, SCP should be keeping requirements.txt and setup.py in this repo up to date so it is valid against production for manage_study.py) to git pull against this repo and pip install -r requirements.txt. 

If there are other, less hack-y solutions to ensure that the user is running the appropriate version of ingest code, that would be even better.

Comment (knapii-developments) [reply scripts/manage_study.py:158]: @eweitz I remember speaking with you about this issue. Do you mind commenting on this.

Comment (eweitz) [reply scripts/manage_study.py:158]: Per our group chat just now, I think we've got a plan to address the issues Jean notes.

Comment (knapii-developments) [reply scripts/manage_study.py:167]: [SCP-2415](https://broadworkbench.atlassian.net/browse/SCP-2415) and [SCP-2414](https://broadworkbench.atlassian.net/browse/SCP-2414) track these issues

File: scripts/cli_parser.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
+import argparse
+
```
Comment (devonbush) [inline scripts/cli_parser.py:2]: add high-level comment about what this script is for

File: scripts/manage_study.py [hunk 3]
```diff
@@ -140,370 +146,47 @@ def download_from_bucket(file_path):
...
-    report_issues(metadata)
-    exit_if_errors(metadata)
+    study_name = get_parsed_args().study_name
+    dry_run = get_parsed_args().dry_run
+    verbose = get_parsed_args().verbose
+    study_accession_res =get_connection().get_study_attribute(
+        study_name=study_name,
+        attribute='accession',
+        dry_run=dry_run)
+    study_file = ObjectId('addedfeed000000000000000')
```
Comment (devonbush) [inline scripts/manage_study.py:156]: are these just mock ids, or do they need to correspond to an actual study in the environment?  add a comment either way explaining

Comment (knapii-developments) [reply scripts/manage_study.py:157]: 9fe8324

File: scripts/cli_parser.py (hunk 2 above)
Comment (knapii-developments) [reply scripts/cli_parser.py:6]: 9fe8324

**Pull request merged**